File: VB/WindowsApplication1/CustomEditor/MyPainter.vb
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1

    <System.ComponentModel.DesignerCategory("")>
    Public Class MyPainter
        Inherits ButtonEditPainter

        Public Sub New()
        End Sub

        Protected Overrides Sub DrawContent(ByVal info As ControlGraphicsInfoArgs)
            MyBase.DrawContent(info)
        End Sub

        Protected Overrides Sub DrawText(ByVal info As ControlGraphicsInfoArgs)
            Call RichTextEditPainter.DrawRTF(GetViewInfo(info), info.Cache)
        End Sub

        Private Function GetViewInfo(ByVal info As ControlGraphicsInfoArgs) As RichTextEditViewInfo
            Dim ri As RepositoryItemMyButtonEdit = TryCast(TryCast(info.ViewInfo, ButtonEditViewInfo).Item, RepositoryItemMyButtonEdit)
            Dim vi As RichTextEditViewInfo = New RichTextEditViewInfo(ri.RichEdit)
            vi.Bounds = info.ViewInfo.ContentRect
            vi.EditValue = info.ViewInfo.DisplayText
            vi.CalcViewInfo(info.Graphics)
            Return vi
        End Function
    End Class
End Namespace

File: CS/WindowsApplication1/CustomEditor/MyPainter.cs
using System;
using System.ComponentModel;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Repository;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyPainter : ButtonEditPainter
    {

        public MyPainter()
        {

        }

        protected override void DrawContent(ControlGraphicsInfoArgs info)
        {
            base.DrawContent(info);
        }

        protected override void DrawText(ControlGraphicsInfoArgs info)
        {
            RichTextEditPainter.DrawRTF(GetViewInfo(info), info.Cache);
        }
        private RichTextEditViewInfo GetViewInfo(ControlGraphicsInfoArgs info)
        {
            RepositoryItemMyButtonEdit ri = (info.ViewInfo as ButtonEditViewInfo).Item as RepositoryItemMyButtonEdit;
            RichTextEditViewInfo vi = new RichTextEditViewInfo(ri.RichEdit);
            vi.Bounds = info.ViewInfo.ContentRect;
            vi.EditValue = info.ViewInfo.DisplayText;
            vi.CalcViewInfo(info.Graphics);
            return vi;

        }
    }

}
